olp-cpp-sdk-core/include/olp/core/network2/Network.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <functional>
+
+#include <olp/core/CoreApi.h>
+
+#include <olp/core/network2/NetworkRequest.h>
+#include <olp/core/network2/NetworkResponse.h>
+#include <olp/core/network2/NetworkTypes.h>
+
+namespace olp {
+namespace network2 {
+/**
+ * @brief The Network class represents HTTP client abstraction
+ */
+class CORE_API Network {
+ public:
+  /**
+   * @brief NetworkCallback represents callback to be called when request
+   * processed or cancelled
+   */
+  using Callback = std::function<void(NetworkResponse response)>;
+
+  /**
+   * @brief HeaderCallback represents callback to be called when a header has
```

Comment:
It is not mandatory to use alias/class/method name in doxygen comment. Doxygen will associate this correctly.